native/core/src/activities.rs

```diff
@@ -123,7 +124,18 @@ impl Activity {
 
     pub fn type_str(&self) -> String {
         match &self.inner {
-            ActivityContent::MembershipChange(c) => c.as_str(),
+            ActivityContent::MembershipChange(c) => {
+                return c.change();
+            }
+            ActivityContent::ProfileChange(c) => {
+                if c.display_name_change().is_some() {
+                    "displayName"
+                } else if c.avatar_url_change().is_some() {
+                    "avatarUrl"
+                } else {
+                    unreachable!()
```

Comment:
why? 
```suggestion
```

Reply:
Do you want to return empty string instead?